### ISSUE TITLE
Specify default environment

### ISSUE THREAD

**Issue by delasteve** [angular/angular-cli#1923]
I have added a few different environments configs for each of the different ones we have, including a `local` config. I would like to specify the `local` config to be the default and not dev. Currently, I can get around that by `ng build --env=local`.

It's minor, but it would be a nice improvement 😄 

Thanks!

**Comment by bobrosoft** [comment]
You can use symlink to you `local` and name it `environment.dev.ts`

**Comment by delasteve** [comment]
My company has a server called dev, though. Otherwise I would.

**Comment by rajatrocks37** [comment]
Hello,

Is this feature added or not ?

**Comment by mionick** [comment]
My team has a server setup with DEV, UAT, and PROD servers, as well as option to run backend locally. 
In our workflow, running locally is the desired default option.

This feature would be greatly appreciated, so that we don't have to specify that developers remember to run ng serve --env=local each time.

**Comment by ninjamyst** [comment]
I am facing the same issue and would really like the ability to specific the default environment to be local instead of dev.  Our "dev" environment is a separate dev build environment on its own server.  Yes, we can use ng serve --env=local but again, it often leads to confusion with developers or wasted time wondering why their changes are not showing up.  

We "could" also configure the angular-cli.json file so "dev" points to the local environment file but that just feels very hacky, especially when our "dev" builds should be using dev environment file.

**Comment by zyv** [comment]
In our case, we are developing an app that has different build configurations per client where certain parts are turned on & off via feature flags / styling changes are applied at build time. So our configurations look like:

- `own-local`
- `own-dev`
- `own-stage`
- `own-prod`

- `client1-local`
- `client1-dev`
- `client1-stage`
- `client1-prod`

In this setting, if the environment is not given at all, building whatever there is to build just doesn't make any sense. So ideally we would like to be able to make specifying the environment mandatory via a config option.

However, next best solution would be to have the possibility to specify a default environment to at least achieve predictable behavior instead of Angular CLI taking `environment.ts` by default and building some unusable rubbish...

@pchorus

**Comment by bhaskerchari** [comment]
@clydin ,

I don't think it still addresses the original problem mentioned in this post even in Angular CLI 6.0+ 
I am not sure if you have really validated the use case that has been asked here.

So how do we make sure to trigger ng serve --env=local by default simply by just running ng serve ?

Did any one else noticed the same or I am missing something? Please share your thoughts.

**Comment by filipesilva** [comment]
@bhaskerchari there is no `--env` flag anymore in VLI v6+. Instead, there is a `fileReplacements` property in your `angular.json` file:
```
 // ...
          "configurations": {
            "production": {
              "fileReplacements": [
                {
                  "replace": "src/environments/environment.ts",
                  "with": "src/environments/environment.prod.ts"
                }
              ],
 // ...
```

This means that on production builds, the `src/environments/environment.ts` file will be replaced with `src/environments/environment.prod.ts`. You can name this file anything you want. It is a direct file replacement in the build pipeline. You can also replace other files.

If you want your file to be called `environment.local.ts` you can do that, and then replace it in production builds.
